### ISSUE TITLE
Dependency to "musl" on Linux undocumented

### ISSUE THREAD

**Issue by uablrek** [ligato/networkservicemesh#601]
On Ubuntu 18.04 there is a dependency to `musl`. Install with;

```
sudo apt install musl
```

The symptom is **very** confusing for those not familiar with dynamic loading and loaders;
```
> ls /home/userid/go/bin/deepcopy-gen
/home/userid/go/bin/deepcopy-gen*
> /home/userid/go/bin/deepcopy-gen -h
bash: /home/userid/go/bin/deepcopy-gen: No such file or directory
```
The file is there but when you run it `bash` says that it isn't. It it the `musl` loader that is missing.

**Comment by edwarnicke** [comment]
Could you say more?  I don’t quite follow.

**Comment by uablrek** [comment]
To build on Ubuntu the "deepcopy-gen" is (built and?) used in the process. But it is built towards "musl", the libc used by "Alpine", probably in some docker container (I don't know). The point is that building NSM fails since " deepcopy-gen" is called by the build tools, and fails.

So it is not NSM that really has a dependency to "musl" but it **must** be installed on Ubuntu to make the build process work.

**Comment by edwarnicke** [comment]
Ah... OK... I just tried to build on fresh ubuntu xenial container and did not hit a musl dependency....

That said... I hit things that could easily cause problems (which might eventually end in the musl issue if one were persistent).

So... a couple of questions...

What version of Go do you have:

```
go version
```

I ask because doing ```apt-get golang``` installs golang 1.6.2 ... well pre-vendor support, and we *do* use vendor support.

Did you try to install deepcopy-gen with

```
go install ./vendor/k8s.io/code-generator/cmd/deepcopy-gen/
```

or with 
```
go get k8s.io/code-generator/cmd/deepcopy-gen
```

Note: the former is how we do it in CI... but our docs should be updated to reflect that.  You seem diligent enough you might have gone and looked at the .circleci/config.yml and figured that bit out, which is why I'm asking :)

I appreciate your patience and diligence on all of this... it will result in updates to our docs which will hopefully help others... your issue is *interesting* and I'd like to get to the bottom of it :)

**Comment by edwarnicke** [comment]
@uablrek Does this docs update on build instructions help:

https://github.com/ligato/networkservicemesh-site/pull/60

**Comment by uablrek** [comment]
You are right, when I build deepcopy-gen as you suggest it builds with the standard `libc` and works fine.

I don't know why I got this problem. I built just by pasting your build instructions and checked the date on `$GOPATH/bin/deepcopy-gen` to see if it was build at the time I tried to build NSM, and it was.

Anyway I got around it anyway since I have seen loader problems before, but I remember my total confusion the first time so I thought an issue was in place to save others from that.

But I now updated and re-built from scratch (I hope) and did not get the problem so I close this issue.

If some one else encounters it you know what they talk about at least.

Thanks for your time.
